### ISSUE TITLE
Model.copy() is not working properly

### ISSUE THREAD

**Issue by yashtrihbti** [OpenMined/PySyft#2207]
**windows 10
python 3.7.3
pytorch=1.0.1
pysyft=0.1.13**
```
#Train Bob's and Alice's Models (in parallel) 

client_copy=[0]*2;
for epoch in range(1, 2):
    model.train()
    for batch_idx, (data, target) in enumerate(federated_train_loader): # <-- now it is a distributed dataset
        if batch_idx==0:
            bobmodel=model.copy().send(data.location) # <-- NEW: send the model to the rig
        data, target = data.to(device), target.to(device)
        optimizer.zero_grad()
        output = bobmodel(data)
        loss = F.nll_loss(output, target)
        loss.backward()
        optimizer.step() 
        if batch_idx % args.log_interval == 0:
            loss = loss.get() # <-- NEW: get the loss back
            print('Train Epoch: {} [{}/{} ({:.0f}%)]\tLoss: {:.6f}'.format(
                epoch, batch_idx * args.batch_size, len(federated_train_loader) * args.batch_size,
                100. * batch_idx / len(federated_train_loader), loss.data))
```

Give output:-
```
Train Epoch: 1 [0/60032 (0%)]	Loss: 2.305134
Train Epoch: 1 [1920/60032 (3%)]	Loss: 2.306227
Train Epoch: 1 [3840/60032 (6%)]	Loss: 2.314577
Train Epoch: 1 [5760/60032 (10%)]	Loss: 2.301752
Train Epoch: 1 [7680/60032 (13%)]	Loss: 2.304796
Train Epoch: 1 [9600/60032 (16%)]	Loss: 2.305816
Train Epoch: 1 [11520/60032 (19%)]	Loss: 2.307251
Train Epoch: 1 [13440/60032 (22%)]	Loss: 2.305082
Train Epoch: 1 [15360/60032 (26%)]	Loss: 2.309973
Train Epoch: 1 [17280/60032 (29%)]	Loss: 2.296431
Train Epoch: 1 [19200/60032 (32%)]	Loss: 2.309648
Train Epoch: 1 [21120/60032 (35%)]	Loss: 2.313736
Train Epoch: 1 [23040/60032 (38%)]	Loss: 2.303707
Train Epoch: 1 [24960/60032 (42%)]	Loss: 2.308319
Train Epoch: 1 [26880/60032 (45%)]	Loss: 2.306007
Train Epoch: 1 [28800/60032 (48%)]	Loss: 2.307840
```

While without making copy of model:-
 `bobmodel=model.send(data.location)`
give output 
```
Train Epoch: 1 [0/60032 (0%)]	Loss: 2.305134
Train Epoch: 1 [1920/60032 (3%)]	Loss: 2.156802
Train Epoch: 1 [3840/60032 (6%)]	Loss: 1.896611
Train Epoch: 1 [5760/60032 (10%)]	Loss: 1.440396
Train Epoch: 1 [7680/60032 (13%)]	Loss: 0.867146
Train Epoch: 1 [9600/60032 (16%)]	Loss: 0.654510
Train Epoch: 1 [11520/60032 (19%)]	Loss: 0.593326
Train Epoch: 1 [13440/60032 (22%)]	Loss: 0.455695
Train Epoch: 1 [15360/60032 (26%)]	Loss: 0.370735
Train Epoch: 1 [17280/60032 (29%)]	Loss: 0.303775
Train Epoch: 1 [19200/60032 (32%)]	Loss: 0.312882
Train Epoch: 1 [21120/60032 (35%)]	Loss: 0.369826
Train Epoch: 1 [23040/60032 (38%)]	Loss: 0.237517
Train Epoch: 1 [24960/60032 (42%)]	Loss: 0.187326
Train Epoch: 1 [26880/60032 (45%)]	Loss: 0.522530
Train Epoch: 1 [28800/60032 (48%)]	Loss: 0.225577
 ```

Model.copy() is not working properly. As loss is not decreasing in same input with model.copy while it is decreasing without copy in starting even before federated averaging.

**Comment by LaRiffle** [comment]
Hey, if you have the whole code I'd love to see it to better understand our use case. Should I infer that all the data is owned by bob, as you only the model once per epoch and to bob?

Also, you should update `pytorch=1.0.1 -> pytorch=1.1` and `pysyft=0.1.13 -> pysyft=0.1.15a1`; many changes are happening currently :)

**Comment by yashtrihbti** [comment]
[`Link`](https://drive.google.com/file/d/1CnvBwgLMd4W1jtXlT2PBRuGjWQBmHpV0/view?usp=sharing)
If you change this line "bobmodel=model.send(data.location)" to "bobmodel=model.copy().send(data.location)" then loss is not decreasing.

**Comment by yashtrihbti** [comment]
Run the whole code by replacing "bobmodel=model.send(data.location)" with "bobmodel=model.copy().send(data.location)" you see that loss is not decreasing for latter case.

**Comment by LaRiffle** [comment]
Oh yes of course!
You have given to your optimizer the parameters of the original model, not of the copy, thats why the copy of the model doesn't improve.
Another thing is that during an epoch, you only send once the model to the first worker, so when you will have data batches owned by the other worker, it will fail.

I believe this Issue is solved, if you have other issues related to your use case, please let us know!
